employees/views.py

```diff
@@ -220,3 +222,21 @@ def form_valid(self, form: AdminReportForm) -> HttpResponseRedirectBase:
         self.object.editable = True
         self.object.save()
         return super().form_valid(form)
+
+
+@method_decorator(login_required, name="dispatch")
+class ProjectReportList(ListView):
```

Comment:
After looking at this finally, I think you were right in the version - it should be `DetailView`, with model Project, and reports should taken in template by some Project model method to apply ordering. This method could look like this:

```
def get_reports_ordered(self):
    return self.report_set.order_by("-date", "project__name")
```

Please only remember to have it already fetched using `.select_related` in `get_queryset()` method.

Reply:
Done